### ISSUE TITLE
YOLO26n: `model.predict()` and `model.val()` produce inconsistent detection results

### ISSUE THREAD

**Issue by soanduong** [ultralytics/ultralytics#24452]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am using pretrained YOLO26n model to test on a single image `000000027620.jpg` from the COCO validation set. The prediction result contains one false positive: the model recognizes the `bottle` as `cup`. Below are the inference code and prediction results

```python
# Run inference
results = model(img_path, imgsz=640, conf=0.1, iou=0.7,
                augment=False, verbose=False)

# Get plotted result image
result_img = results[0].plot()

# Convert BGR -> RGB for matplotlib
result_img_rgb = cv2.cvtColor(result_img, cv2.COLOR_BGR2RGB)

# Display result
plt.figure(figsize=(10, 10))
plt.imshow(result_img_rgb)
plt.axis("off")
plt.show()
``` 

_**Prediction result:**_
<img width="786" height="591" alt="Image" src="https://github.com/user-attachments/assets/845f162d-bf79-499b-8869-111b2627b350" />

I then placed the image and its corresponding ground-truth annotation into a folder and evaluated the model using model.val(). However, the validation result is different:
_**Validation results**_
<img width="1161" height="260" alt="Image" src="https://github.com/user-attachments/assets/32de0b8c-1412-4a9e-8cc5-d9a3f3fe3f43" />

From the validation metrics, the `bottle` class has `precision=1` and `recall=1`, which suggests that `model.val()` correctly recognizes `bottle`.

Could anyone please explain what/where I might be misunderstanding or doing correctly? Any suggestions or insights would be greatly appreciated.

Thank you very much!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @soanduong, thank you for your detailed report and for using Ultralytics 🚀! This is an automated response to help gather the right debugging details quickly, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).
For this specific report, it would be especially helpful if you could share:
- The exact `model.val()` call you used
- The dataset YAML or folder structure used for validation
- The exact `ultralytics` package version
- Whether the same image size, confidence, and IoU settings were used across prediction and validation
- A small [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) showing the same image, annotation, and commands used for both inference and validation 🔍
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by soanduong** [comment]
I would like to provide the structure of the data used for validation:

<img width="292" height="387" alt="Image" src="https://github.com/user-attachments/assets/132bac26-cb91-4af0-af7c-68034caf681b" />

And the configuration for for evaluation:
[data.yaml](https://github.com/user-attachments/files/27594062/data.yaml)

**Comment by Y-T-G** [comment]
Duplicate of https://github.com/ultralytics/ultralytics/issues/2783

**Comment by Y-T-G** [comment]
Hello.

You can read this: https://github.com/ultralytics/ultralytics/issues/2783#issuecomment-2413530537

**Comment by soanduong** [comment]
> Hello.
> 
> You can read this: [#2783 (comment)](https://github.com/ultralytics/ultralytics/issues/2783#issuecomment-2413530537)

Thank you very much!

**Comment by banu4prasad** [comment]
training the model `yolo train` default value `rect=False` but while validating `yolo val`default value changes to `rect=True`

when validating the model on visdrone test split i found 

when rect=True map50=0.327
when rect=False map50=0.325

so imo if user trains the model with rect=True then he needs to val with rect=True to get similar eval metrics

**Comment by soanduong** [comment]
> training the model `yolo train` default value `rect=False` but while validating `yolo val`default value changes to `rect=True`
> 
> when validating the model on visdrone test split i found
> 
> when rect=True map50=0.327 when rect=False map50=0.325
> 
> so imo if user trains the model with rect=True then he needs to val with rect=True to get similar eval metrics

Thank you very much. That saved my day!
Besides, I found that I need to set both `conf` and `iou` to the same values to get the consistent results between `model.eval()` and `model.predict()`.

**Comment by glenn-jocher** [comment]
Correct, and just one small note: for a true 1:1 comparison use `model.val()` (not `model.eval()`) with the same `conf`, `iou`, `imgsz`, `rect`, and ideally `batch=1`, since [predict padding behavior](https://docs.ultralytics.com/modes/predict#fixed-shape-vs-minimum-rectangle-rect) can otherwise change detections slightly.
